timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLock.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
```

Comment:
Sounds fair. I am still debating what the best path forward here is on the API switch. The internal transaction protocol can switch to the new lock service API immediately, but we are still exposing a `TransactionManager#runTaskWithLocksWithRetry` method that relies on the old lock service - which some internal products use.

The easiest path forward (optimizing on ease of migrating users) would be to actually run the existing lock service in parallel. Otherwise we'll need to either support everything users currently do via that API, or get them to change their usage of it. I think currently the only thing this PR lacks that we might need is bounded blocking time.